lib/raidenclient/RaidenClient.ts

```
@@ -103,51 +133,86 @@ class RaidenClient extends BaseClient {
+   * @param payload The token swap payload
    * @param identifier An identification number for this swap
    */
-  public async tokenSwap(target_address: string, payload: TokenSwapPayload, identifier: string): Promise<string> {
-    let endpoint: string = `token_swaps/${target_address}`;
+  public async tokenSwap(target_address: string, payload: TokenSwapPayload, identifier: string): Promise<void> {
+    let endpoint = `token_swaps/${target_address}`;
     if (identifier) {
       endpoint += `/${identifier}`;
     }
 
-    const res: any = await this.sendRequest(endpoint, 'PUT', payload);
-    if (res.statusCode === 201) {
-      return 'swap created';
-    } else {
-      throw new Error(`${res.statusCode}: ${res.statusMessage}`);
-    }
+    const res = await this.sendRequest(endpoint, 'PUT', payload);
+    assert(res.statusCode === 201, `${res.statusCode}: ${res.statusMessage}`);
   }
 
   /**
    * Get info about a given raiden payment channel.
+   * @param channel_address The address of the channel to query
    */
-  public async getChannelInfo(channel_address: string): Promise<ChannelInfo> {
-    assert(typeof channel_address === 'string', 'channel_address must be a string');
+  public getChannel = async (channel_address: string): Promise<Channel> => {
+    const endpoint = `channels/${channel_address}`;
+    const res = await this.sendRequest(endpoint, 'GET');
+    assert(res.statusCode === 200, `${res.statusCode}: ${res.statusMessage}`);
+    return parseResponseBody(res);
+  }
 
-    const endpoint: string = `channels/${channel_address}`;
-    const res: any = await this.sendRequest(endpoint, 'GET', undefined);
+  /**
+   * Get info about all non-settled channels.
+   */
+  public getChannels = async (): Promise<[Channel]> => {
+    const endpoint = 'channels';
+    const res = await this.sendRequest(endpoint, 'GET');
+    assert(res.statusCode === 200, `${res.statusCode}: ${res.statusMessage}`);
```

I think we should throw an operational error (defined in the `errors` files), and keep the `assert` statements for scenarios that are unexpected, and ideally should never happen (unlike operational errors, which are expected to happen).

And if we're not interested in failing the entire process when that happens, we should remember to always wrap these RaidenClient calls in `try/catch` clause.